specs/experimental/op-contracts-manager.md

```diff
@@ -153,27 +166,119 @@ This provides the following benefits:
 
 - Contract addresses for a chain can be derived as a function of chain ID without any RPC calls.
 - Chain ID uniqueness is enforced for free, as a deploy using the same chain ID
-will result in attempting to deploy to the same address, which is prohibited by
-the EVM.
+  will result in attempting to deploy to the same address, which is prohibited by
+  the EVM.
   - This property is contingent on the proxy and `AddressManager` code not
-  changing when OP Contracts Manager is upgraded. Both of these are not planned to
-  change.
+    changing when OP Contracts Manager is upgraded. Both of these are not planned to
+    change.
   - The OP Contracts Manager is not responsible for enforcing chain ID uniqueness, so it is acceptable
-  if this property is not preserved in future versions of the OP Contracts Manager.
+    if this property is not preserved in future versions of the OP Contracts Manager.
+
+## Upgrading
+
+### Interface
+
+#### `upgrade`
+
+The `upgrade` method is used by the Upgrade Controller to upgrade the full set of L1 contracts for
+all chains that it controls.
+
+It has the following interface:
+
+```solidity
+function upgrade(ISystemConfig[] _systemConfigs, NewChainConfig[] _newConfigs) public;
+```
+
+For each chain successfully upgraded, the following event is emitted:
+
+```solidity
+event Upgraded(uint256 indexed l2ChainId, SystemConfig indexed systemConfig);
```

Comment:
This doesn't match the `Deployed` event very closely, which is: 

```
event Deployed(
        uint256 indexed outputVersion, uint256 indexed l2ChainId, address indexed deployer, bytes deployOutput
    );
```

Open questions I have: 
- Is there a need for an `upgradeOutput`? What would go in it?
- Is there any benefit to including the `upgrader` address, since it's a DELEGATECALL it will actually be emitted by that address.
- Is it helpful to emit both `l2ChainId` and `systemConfig`?

Reply:
1. Not sure we would need `upgradeOutput` on the basis that all the information you need about an upgrade should be implicit from the OPCM upgrade function itself. 
2. I see your point that it'll be emitted by the upgrader. I don't see any downside of putting it into the event log though. 
3. I think so.

Reply:
1. Agreed we don't need `upgradeOutput`, unsure what that would include
2. Agreed we may as well include this address, it makes it easier to retrieve the address without having to trace to see the address that emitted that event
3. Seems like we may as well, this could be useful for running post-checks to analyze logs to make sure all expected chains were ugpraded

Reply:
```suggestion
event Upgraded(uint256 indexed l2ChainId, SystemConfig indexed systemConfig, address indexed upgrader);
```